src/odemis/acq/stream/_sync.py

```diff
@@ -1741,30 +1913,48 @@ def _runAcquisition(self, future):
             pos_flat = spot_pos.reshape((-1, 2))  # X/Y together (X iterates first)
             rep = self.repetition.value
             self._acq_data = [[] for _ in self._streams]  # just to be sure it's really empty
+            self._live_data = [[] for s in self._streams]
             self._raw = []
             self._anchor_raw = []
             logging.debug("Starting e-beam sync acquisition with components %s",
                           ", ".join(s._detector.name for s in self._streams))
-
             tot_num = numpy.prod(rep)
 
             # initialize leeches
             leech_np, leech_time_ppx = self._startLeeches(px_time, tot_num, (rep[1], rep[0]))
 
+            self._im_needs_recompute = threading.Event()
+            self._live_update_period = 0.33 # s
+            self._liveImageThread = threading.Thread(target = self._image_thread,
+                                                     args = (weakref.ref(self), self._live_update_period),
+                                                     name = "Live update of the acquisition")
+            self._liveImageThread.daemon = True
+            self._liveImageThread.start()
+
             # number of spots scanned so far
             spots_sum = 0
             while spots_sum < tot_num:
+                px_idx = (spots_sum//rep[0], spots_sum%rep[0]) #current pixel index
```

Comment:
Move this line to just before `_current_scan_area`.
BTW, please add spaces around the operators.